src/libraries/System.Collections.Concurrent/tests/ConcurrentStackTests.cs

```diff
@@ -63,6 +63,26 @@ public void PushRange_NoItems_NothingAdded()
             Assert.True(s.IsEmpty);
         }
 
+        [Fact]
+        public void PushRange_NoItems_NothingAdded_EmptyArrayWithRangeSpecified()
+        {
+            var s = new ConcurrentStack<int>();
+            Assert.True(s.IsEmpty);
+
+            s.PushRange(new int[0], 0, 0);
+            Assert.True(s.IsEmpty);
+        }
+
+        [Fact]
+        public void PushRange_NoItems_NothingAdded_EmptyArrayNoRangeSpecified()
+        {
+            var s = new ConcurrentStack<int>();
+            Assert.True(s.IsEmpty);
+
+            s.PushRange(new int[0]);
+            Assert.True(s.IsEmpty);
+        }
```

Comment:
We should also validate with a zero count at the end of a non-empty array,  e.g.
```C#
int[] arr = new int[2];
s.PushRange(arr, arr.Length, 0);
```